storage/storage-interface/src/async_proof_fetcher.rs

```diff
@@ -101,9 +101,10 @@ impl AsyncProofFetcher {
                     .verify_by_hash(root_hash, state_key.hash(), value_hash)
                     .map_err(|err| {
                         anyhow!(
-                            "Proof is invalid for key {:?} with state root hash {:?}: {}.",
+                            "Proof is invalid for key {:?} with state root hash {:?}, at version {}: {}.",
```

Comment:
@gelash brought up a good point that we should print the proof itself, and/or even check the hashes add up, if the failure is about the existence of the leaf.